src/Page.php

```diff
@@ -440,6 +441,48 @@ private function waitForReloadGenerator($eventName, $loaderId)
         }
     }
 
+    /**
+     * Wait until page contains Node.
+     *
+     * @param string $selectors
+     * @param int    $timeout
+     *
+     * @throws Exception\OperationTimedOut
+     */
+    public function waitUntilContainsElement(string $selectors, int $timeout = 30000): self
+    {
+        $this->assertNotClosed();
+
+        Utils::tryWithTimeout($timeout * 1000, $this->waitForElement($selectors));
+
+        return $this;
+    }
+
+    /**
+     * @return bool|\Generator
+     *
+     * @internal
+     */
+    public function waitForElement(string $selectors, int $position = 1)
```

Comment:
should probably add a timeout option to prevent infinite loot if the element never appear